gravitee-am-service/src/main/java/io/gravitee/am/service/http/WebClientInitializer.java

```diff
@@ -0,0 +1,129 @@
+            setDelegate(delegate);
+        }
+
+        void setDelegate(io.vertx.ext.web.client.WebClient delegate){
+            this.delegate = delegate;
+        }
+    }
+
+    private static class WebClientStub implements io.vertx.ext.web.client.WebClient {
+        private static final WebClientStub INSTANCE = new WebClientStub();
+
+
+        @Override
+        public HttpRequest<Buffer> request(HttpMethod httpMethod, SocketAddress socketAddress, int i, String s, String s1) {
+            throw new UnsupportedOperationException("WebClient is being initialized");
+        }
+
+        @Override
+        public HttpRequest<Buffer> request(HttpMethod httpMethod, SocketAddress socketAddress, int i, String s, UriTemplate uriTemplate) {
+            throw new UnsupportedOperationException("WebClient is being initialized");
+        }
+
+        @Override
+        public HttpRequest<Buffer> request(HttpMethod httpMethod, SocketAddress socketAddress, String s, String s1) {
+            throw new UnsupportedOperationException("WebClient is being initialized");
+        }
+
+        @Override
+        public HttpRequest<Buffer> request(HttpMethod httpMethod, SocketAddress socketAddress, String s, UriTemplate uriTemplate) {
+            throw new UnsupportedOperationException("WebClient is being initialized");
+        }
+
+        @Override
+        public HttpRequest<Buffer> request(HttpMethod httpMethod, SocketAddress socketAddress, String s) {
+            throw new UnsupportedOperationException("WebClient is being initialized");
+        }
+
+        @Override
+        public HttpRequest<Buffer> request(HttpMethod httpMethod, SocketAddress socketAddress, UriTemplate uriTemplate) {
+            throw new UnsupportedOperationException("WebClient is being initialized");
+        }
+
+        @Override
+        public HttpRequest<Buffer> request(HttpMethod httpMethod, SocketAddress socketAddress, RequestOptions requestOptions) {
+            throw new UnsupportedOperationException("WebClient is being initialized");
+        }
+
+        @Override
+        public HttpRequest<Buffer> requestAbs(HttpMethod httpMethod, SocketAddress socketAddress, String s) {
+            throw new UnsupportedOperationException("WebClient is being initialized");
+        }
+
+        @Override
+        public HttpRequest<Buffer> requestAbs(HttpMethod httpMethod, SocketAddress socketAddress, UriTemplate uriTemplate) {
+            throw new UnsupportedOperationException("WebClient is being initialized");
+        }
+
+        @Override
+        public void close() {
+            throw new UnsupportedOperationException("WebClient is being initialized");
+        }
+    }
+
+    private static final class RetryWithDelay implements Function<Flowable<Throwable>, Publisher<?>> {
+        private int delayInSec = 1;
+
+        @Override
+        public Publisher<?> apply(Flowable<Throwable> throwableFlowable) {
+            return throwableFlowable.flatMap(err -> {
```

Comment:
should it retry indefinitely? I think it could be a good idea to limit it and return an error when the limit is exceeded. Please look at this: 

  .retryWhen(throwableFlowable -> throwableFlowable
                        .flatMap(err -> {
                            if (counter.getAndIncrement() < 50) {
                                log.info("Retry initialize pools, retry: {}", counter.get());
                                long oldValue = delay.get();
                                long newValue = oldValue < 60 ? oldValue * 2L : oldValue;
                                delay.set(newValue);
                                return Flowable.timer(newValue, TimeUnit.SECONDS);
                            } else {
                                log.info("Retry limit exceeded to initialize pools");
                                return Flowable.error(err);
                            }
                        }))